csp_billing_adapter/adapter.py

```diff
@@ -214,7 +216,23 @@ def main() -> None:
             log
         )
 
-        initial_adapter_setup(pm.hook, config, log)
+        cache, csp_config = initial_adapter_setup(pm.hook, config, log)
+
+        try:
+            # Test metering API access
+            process_metering(
+                config,
+                cache,
+                pm.hook,
+                empty_metering=True,
+                dry_run=True
+            )
+        except Exception as error:
+            raise CSPBillingAdapterException(
+                f'Failed to validate metering API access: {error}'
```

Comment:
Given the exception handing currently present in process_metering() this exception block will only trigger if we have issues updating the csp_config (in the meter_billing() failed exception handling) or cache (in the meter_billing() succeeded code flow) so the text of this exception message may be a little bit ambiguous...

It will not actually trigger if we had a failed meter_billing() but successfully update the csp_config to reflect that.

So perhaps the following may be more appropriate?

```suggestion
                f'Fatal error while validating metering API access: {error}'
```

Reply:
Yeah, that falls into #79

Reply:
I'll update the error string per comment in this PR. It will probably change with the next PR.